mcuboot build fails with nrf52 internal RC oscillator

When I try to build my app with custom nrf52832 board (that has only internal RC oscillator configured as a LF clock source) it works. But when I try to build mcuboot for the same board, it fails.
Also mentioned here - https://devzone.nordicsemi.com/f/nordic-q-a/81418/bug-in-ncs-v1-7-1-with-32khz-rc-oscillator-and-mcuboot-enabled

Build of my app (I'm using T2 project structure):
```bash
user@6b89f0d24bcb:/workspaces/zephyr_workspace$ west build -b nrf52832_hangy -s hangy_firmware
-- west build: generating a build system
Including boilerplate (Zephyr base): /workspaces/zephyr_workspace/zephyr/cmake/app/boilerplate.cmake
-- Application: /workspaces/zephyr_workspace/hangy_firmware
-- Zephyr version: 2.7.0 (/workspaces/zephyr_workspace/zephyr), build: zephyr-v2.7.0
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: nrf52832_hangy
-- Cache files will be written to: /home/user/.cache/zephyr
-- Using toolchain: zephyr 0.13.2 (/opt/toolchains/zephyr-sdk-0.13.2)
-- Found dtc: /opt/toolchains/zephyr-sdk-0.13.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found BOARD.dts: /workspaces/zephyr_workspace/hangy_firmware/boards/arm/nrf52832_hangy/nrf52832_hangy.dts
-- Found devicetree overlay: /workspaces/zephyr_workspace/hangy_firmware/boards/nrf52832_hangy.overlay
-- Generated zephyr.dts: /workspaces/zephyr_workspace/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /workspaces/zephyr_workspace/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /workspaces/zephyr_workspace/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /workspaces/zephyr_workspace/build/zephyr/dts.cmake
Parsing /workspaces/zephyr_workspace/zephyr/Kconfig
Loaded configuration '/workspaces/zephyr_workspace/hangy_firmware/boards/arm/nrf52832_hangy/nrf52832_hangy_defconfig'
Merged configuration '/workspaces/zephyr_workspace/hangy_firmware/prj.conf'
Configuration saved to '/workspaces/zephyr_workspace/build/zephyr/.config'
Kconfig header saved to '/workspaces/zephyr_workspace/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /workspaces/zephyr_workspace/build
-- west build: building application
[1/276] Preparing syscall dependency handling

[269/276] Linking C executable zephyr/zephyr_prebuilt.elf

[276/276] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:      182568 B       200 KB     89.14%
            SRAM:       33956 B        64 KB     51.81%
        IDT_LIST:          0 GB         2 KB      0.00%
```

Build of mcuboot:
```bash
user@6b89f0d24bcb:/workspaces/zephyr_workspace$ west build -b nrf52832_hangy -d build/nrf52832_hangy_bootloader bootloader/mcuboot/boot/zephyr -- -DBOARD_ROOT=/workspaces/zephyr_workspace/hangy_firmware
-- west build: generating a build system
Including boilerplate (Zephyr base): /workspaces/zephyr_workspace/zephyr/cmake/app/boilerplate.cmake
-- Application: /workspaces/zephyr_workspace/bootloader/mcuboot/boot/zephyr
-- Zephyr version: 2.7.0 (/workspaces/zephyr_workspace/zephyr), build: zephyr-v2.7.0
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: nrf52832_hangy
-- Cache files will be written to: /home/user/.cache/zephyr
-- Using toolchain: zephyr 0.13.2 (/opt/toolchains/zephyr-sdk-0.13.2)
-- Found dtc: /opt/toolchains/zephyr-sdk-0.13.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found BOARD.dts: /workspaces/zephyr_workspace/hangy_firmware/boards/arm/nrf52832_hangy/nrf52832_hangy.dts
-- Found devicetree overlay: /workspaces/zephyr_workspace/bootloader/mcuboot/boot/zephyr/dts.overlay
-- Generated zephyr.dts: /workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/dts.cmake
Parsing /workspaces/zephyr_workspace/bootloader/mcuboot/boot/zephyr/Kconfig
Loaded configuration '/workspaces/zephyr_workspace/hangy_firmware/boards/arm/nrf52832_hangy/nrf52832_hangy_defconfig'
Merged configuration '/workspaces/zephyr_workspace/bootloader/mcuboot/boot/zephyr/prj.conf'
Configuration saved to '/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/.config'
Kconfig header saved to '/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
MCUBoot bootloader key file: /workspaces/zephyr_workspace/bootloader/mcuboot/root-rsa-2048.pem
-- Configuring done
-- Generating done
-- Build files have been written to: /workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader
-- west build: building application
[1/251] Preparing syscall dependency handling

[244/251] Linking C executable zephyr/zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf zephyr/zephyr_prebuilt.map 
: && ccache /opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc   zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj -o zephyr/zephyr_prebuilt.elf  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_prebuilt.cmd  -Wl,-Map=/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/zephyr_prebuilt.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/soc/arm/common/cortex_m/libsoc__arm__common__cortex_m.a  zephyr/soc/arm/nordic_nrf/nrf52/libsoc__arm__nordic_nrf__nrf52.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/sensor/nrf5/libdrivers__sensor__nrf5.a  zephyr/drivers/flash/libdrivers__flash.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/hal_nordic/nrfx/libmodules__hal_nordic__nrfx.a  modules/mbedtls/libmodules__mbedtls.a  modules/mcuboot/libmcuboot_util.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -L"/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/thumb/v7e-m/nofp"  -L/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -mcpu=cortex-m4  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -no-pie  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && cd /workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr && /usr/local/bin/cmake -E echo
/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/nrf5/libdrivers__sensor__nrf5.a(temp_nrf5.c.obj): in function `k_mutex_lock':
/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/syscalls/kernel.h:820: undefined reference to `z_impl_k_mutex_lock'
/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/nrf5/libdrivers__sensor__nrf5.a(temp_nrf5.c.obj): in function `k_sem_take':
/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/syscalls/kernel.h:936: undefined reference to `z_impl_k_sem_take'
/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/nrf5/libdrivers__sensor__nrf5.a(temp_nrf5.c.obj): in function `k_mutex_unlock':
/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/syscalls/kernel.h:836: undefined reference to `z_impl_k_mutex_unlock'
/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/nrf5/libdrivers__sensor__nrf5.a(temp_nrf5.c.obj): in function `k_sem_init':
/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/syscalls/kernel.h:918: undefined reference to `z_impl_k_sem_init'
/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/nrf5/libdrivers__sensor__nrf5.a(temp_nrf5.c.obj): in function `k_mutex_init':
/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/syscalls/kernel.h:802: undefined reference to `z_impl_k_mutex_init'
/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/sensor/nrf5/libdrivers__sensor__nrf5.a(temp_nrf5.c.obj): in function `k_sem_give':
/workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader/zephyr/include/generated/syscalls/kernel.h:953: undefined reference to `z_impl_k_sem_give'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /workspaces/zephyr_workspace/build/nrf52832_hangy_bootloader
```
Possible workaround is to use `CONFIG_CLOCK_CONTROL_NRF_K32SRC_SYNTH=y` option.
My custom board config is basically the nrf52832_mdk with these added options:
```
CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC=y
CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION=y
```

**Environment**
 * OS: Linux
 * zephyr-sdk-0.13.2
 * zephyr 2.7.0, but the same happens with the latest main and also with 2.6.0

--- a/drivers/sensor/nrf5/Kconfig
+++ b/drivers/sensor/nrf5/Kconfig
@@ -13,5 +13,6 @@ config TEMP_NRF5_FORCE_ALT
 config TEMP_NRF5
 	bool "nRF5 Temperature Sensor"
 	depends on HAS_HW_NRF_TEMP
+	depends on MULTITHREADING || TEMP_NRF5_FORCE_ALT
 	help
 	  Enable driver for nRF5 temperature sensor.

